tests/algorithms/pytorch/test_newton_raphson.py

```diff
@@ -109,6 +109,55 @@ def test_train_newton_raphson_shared_states_results(
     assert pytest.approx(np.array(expected_hessian), rel) == shared_states.hessian
 
 
+@pytest.mark.slow  # the test takes ~30s
+def test_hessian_logistic_regression_newton_raphson(torch_algo, seed):
+    """Test that the Hessian matrix is well computed for a logistic regression problem."""
+    np.random.seed(seed)
+
+    n_samples = 300
+    ndim = 100
+    x_train = np.random.randn(n_samples, ndim)
+    y_train = np.random.randint(0, 2, n_samples).reshape(-1, 1)
+
+    class LogisticRegression(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.linear = torch.nn.Linear(ndim, 1)
+
+        def forward(self, x):
+            out = self.linear(x)
+            return out
```

Comment:
```suggestion
def test_hessian_logistic_regression_newton_raphson(perceptron, torch_algo, seed):
    """Test that the Hessian matrix is well computed for a logistic regression problem."""
    np.random.seed(seed)
    n_samples = 30
    ndim = 10
    
    x_train = np.random.randn(n_samples, ndim)
    y_train = np.random.randint(0, 2, n_samples).reshape(-1, 1)

    model = perceptron(ndim, 1)
```

Reply:
The suggestion is not well parametrize, sorry, but hope you get the idea to reuse the `perceptron` fixture defined on top of the file :)